tasks/steps/backup.yml

```diff
@@ -0,0 +1,34 @@
+---
+
+- name: 'Create backups dir'
+  file:
+    path: '{{ cartridge_instances_backups_dir }}'
+    owner: '{{ cartridge_app_user }}'
+    group: '{{ cartridge_app_group }}'
+    state: directory
+    mode: 0660
+  when: inventory_hostname in single_instances_for_each_machine
+
+- name: 'Create instance backup archive'
+  cartridge_backup_instance:
+    instance_id: '{{ instance_info.instance_id }}'
+    stateboard: '{{ stateboard }}'
+    backups_dir: '{{ cartridge_instances_backups_dir }}'
+    console_sock: '{{ instance_info.console_sock }}'
+    instance_conf_file: '{{ instance_info.conf_file }}'
+    app_conf_file: '{{ instance_info.app_conf_file }}'
+  register: backup_res
+
+- name: 'Set "backup_archive_path" and "instance_backup_files" facts'
+  set_fact:
+    backup_archive_path: '{{ backup_res.fact.backup_archive_path }}'
+    instance_backup_files: '{{ backup_res.fact.backup_files }}'
+  when: not backup_res.failed
+
+- name: Fetch backup archive
+  fetch:
+    src: '{{ backup_archive_path }}'
+    dest: '{{ cartridge_fetch_backups_dir }}'
+    flat: true
+  when: cartridge_fetch_backups
```

Comment:
and  when not backup_res.failed?

Reply:
It's not necessary. Playbook will stop after the previous task fails. `when: not backup_res.failed` is checked to prevent failing test with `ignore_errors: true`.

Reply:
But test won't fail, because ignore_errors: true 🙂

What about to remove not backup_res.failed in previous task?